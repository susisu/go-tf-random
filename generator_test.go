package tf_random_test

import (
	"os"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	tf_random "github.com/susisu/go-tf-random"
)

func TestMain(t *testing.M) {
	v := t.Run()
	snaps.Clean(t)
	os.Exit(v)
}

func initTFGen() *tf_random.TFGen {
	g := tf_random.NewTFGen(
		0x00000000_00000000,
		0x89abcdef_01234567,
		0x01234567_89abcdef,
		0xffffffff_ffffffff,
	)
	initCount := 100
	for i := 0; i < initCount; i++ {
		g.Uint32()
	}
	return g
}

func testSnapshot(t *testing.T, g *tf_random.TFGen) {
	numSamples := 100
	seq := make([]uint32, 0, numSamples)
	for i := 0; i < numSamples; i++ {
		seq = append(seq, g.Uint32())
	}
	snaps.MatchSnapshot(t, seq)
}

func TestTFGen_Uint32_snapshot(t *testing.T) {
	g := initTFGen()
	testSnapshot(t, g)
}

func TestTFGen_Split_snapshot(t *testing.T) {
	g1 := initTFGen()
	g2 := g1.Split()
	testSnapshot(t, g1)
	testSnapshot(t, g2)
}

func TestTFGen_Level_snapshot(t *testing.T) {
	g := initTFGen()
	g.Level()
	testSnapshot(t, g)
}

func TestTFGen_SplitN_snapshot(t *testing.T) {
	g1 := initTFGen()
	g2 := g1.SplitN(16, 42)
	testSnapshot(t, g2)
}
