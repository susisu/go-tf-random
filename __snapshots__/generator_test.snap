
[TestTFGen_Uint32_snapshot - 1]
[]uint32{0x9eb0c638, 0x2c8923da, 0xebfbeb8c, 0x5c85d787, 0x5f330d57, 0x8c72ec21, 0x4e5dd417, 0xef281b47, 0x2340aaf9, 0xd70ca050, 0x3d6af200, 0x8fa10920, 0xab55e97e, 0xbf470922, 0x4b058ac9, 0x53826a64, 0x4c061511, 0xb79744c4, 0x6d7c8063, 0x930f66e7, 0x64731bca, 0xa6c3d52c, 0xcb572e8b, 0xc8508591, 0xafc5e986, 0x3e8103f7, 0xd83ddfbd, 0x8266fa22, 0x27b4d545, 0x42956a81, 0x5d87724e, 0x474d8da6, 0xd26243c1, 0x336743fa, 0x6d220023, 0x7fd9583, 0x58d92968, 0x984f7a05, 0x3834a0db, 0xc2859d15, 0xcd5bd243, 0x49c1a601, 0x7425d214, 0x67fb6417, 0x9ac6aa8d, 0xee7db8c, 0xe14471a, 0xddb24ea4, 0x181c4993, 0x50c0347b, 0x6e8b7402, 0xe07c7328, 0x7a482052, 0x40387ea3, 0xb55934e9, 0x74c3f701, 0xe03ce561, 0x76d4db32, 0x4616671, 0xc7b8ef22, 0xd3bec14b, 0xe7bbd065, 0x2f69ab44, 0x88184e42, 0x61dfe41e, 0x24a4012d, 0xe284c75c, 0xba859b92, 0x5a0829b9, 0xa76736cc, 0x6bec2d76, 0xede26849, 0xefe8fd9, 0x53e6f40e, 0xf816ad7c, 0x6e0acb2a, 0x2cc1bab1, 0xbaec582e, 0xf886c51f, 0x6c90ec7c, 0xa1ff77ff, 0xc85be74b, 0x59a1b1a7, 0x447f1bb9, 0x35ff7110, 0x81f0afc8, 0xb73dc083, 0x2eb41b56, 0xea54ac5b, 0x22c80338, 0x761b3309, 0x5041cc89, 0xd8ac9c25, 0x4a6e7646, 0x5e50e0, 0x4cd467d, 0x7b6c3848, 0x95866bc1, 0x61c53468, 0x7cc132d0}
---

[TestTFGen_Split_snapshot - 1]
[]uint32{0x1f47b19d, 0x16b5bdd, 0xdd3fb934, 0x199dda20, 0xd271b549, 0xe37b5d98, 0x7a5a4dde, 0x8055f311, 0xea0179df, 0x4a433ea6, 0x763d6ff1, 0x52832e73, 0xe748d861, 0x45b7884d, 0x1481e4c1, 0xdd4d1e31, 0x71a4e5cc, 0x25950277, 0x29fb870f, 0x418757fa, 0x544519c1, 0x1e1b818d, 0x7da42b46, 0x9530e118, 0x1efa07bf, 0xe9dadea6, 0xf96a0e13, 0xf191f26c, 0xa31917f9, 0x4482322f, 0x3fd72cb9, 0x7ddc70c2, 0x4ce0f22c, 0xd5d69fd5, 0xb06bf8c4, 0xce245401, 0x8fa319cb, 0xd53a2549, 0x8887e524, 0xaa2cb64, 0xe694f0a0, 0x249ddd66, 0x7e02952c, 0xdc4e29a2, 0x98197549, 0x1d4813a4, 0x8a01733a, 0x2f1ca396, 0x2be10a50, 0xc4721e30, 0xd8ebdd12, 0xc0f856ba, 0x3f804cd3, 0xbf8862ad, 0x46858ce1, 0x26da3fec, 0x16d05328, 0xe8c98a5c, 0x2f74999e, 0x4c6cf739, 0xe053c3b5, 0x87ed0c42, 0x9b7e326d, 0xf882011b, 0x14f9b862, 0x4e129d8e, 0x19212213, 0x324dbf78, 0x14d8a692, 0x2648f63b, 0x1764f569, 0x5a2e16, 0x7d8a331c, 0xb70aa49b, 0xb59d834, 0x45581cd0, 0xd61e81b8, 0xcbb42e69, 0xca40864, 0x5a4a43f1, 0xda76d357, 0x22d94dcc, 0xbf1bd921, 0x5872d823, 0xb6d64375, 0xdfb80021, 0xf1a9c60d, 0x4a4e35f2, 0x581742a3, 0xb7bc7443, 0x27f0e905, 0x24ef30c6, 0x91a88b9f, 0xf38660ea, 0x830ea0eb, 0xe4907b31, 0xc0e2f924, 0xd8e94866, 0x81f0a03f, 0x3ca10fd5}
---

[TestTFGen_Split_snapshot - 2]
[]uint32{0x13b02d7e, 0x33420939, 0xccd76b6, 0x9884ae91, 0x8d4fe542, 0xaa89ef8c, 0x4cd7032f, 0x8c432f27, 0x7da38fd2, 0x25ea0a93, 0x42a95f80, 0x26594e29, 0xe8abcf1, 0xe0711c9e, 0x48558ae9, 0xb6823c61, 0xa56ea412, 0x647d0f8c, 0x5c29ced4, 0x81bc6bd1, 0x6f0a1d2b, 0xe813ca58, 0xb94f63a0, 0xec95b5be, 0x4e5aa643, 0x27fa04d0, 0x7bcf79c, 0xf535a2ad, 0xe2565c92, 0x25820b8d, 0xb76834ba, 0xf2b165f7, 0x59a076b1, 0xb21a0fe8, 0xaf14f522, 0x19223dfe, 0x6c62393b, 0xa45ea97b, 0xdc1dc644, 0x10cbc2b3, 0xcda459b2, 0xb0610352, 0xb56e6ae1, 0xb26fda19, 0x248a2897, 0xc0387366, 0xa2f92c1f, 0xc7eb68d5, 0x58610951, 0xf53061ee, 0x8189f45c, 0xec745fc4, 0x9ee02d42, 0x2b53289c, 0xfeeed222, 0x1e258e9b, 0xc8cba431, 0x7015d008, 0x8ee3fce7, 0x65c3c0fa, 0x984f7550, 0x676303a3, 0xae590bba, 0x436ea61b, 0xec5b0a60, 0x27dc5d2a, 0x8e625a6e, 0xeff12479, 0xcb66928c, 0xd07dc292, 0x8e1dbd18, 0x3302bc1f, 0x1571034f, 0x391eab58, 0x4c2efce8, 0x435dd1e0, 0x791e9d88, 0x681e68ec, 0x43f55d81, 0x983bdc46, 0x93b5ec41, 0x77b77c95, 0xe364e5c, 0xecd561ed, 0x2cd9628a, 0xf884b069, 0x2fdeca38, 0xa77b8c9f, 0xf5c8d782, 0x80129297, 0xe0206ea6, 0x5bfb3d09, 0x12986567, 0xe87d24fb, 0x8b82061e, 0x1c9ccdd8, 0x4439b77c, 0x1e8691ff, 0xd52f1255, 0x89fd41d9}
---

[TestTFGen_Level_snapshot - 1]
[]uint32{0x9eb0c638, 0x2c8923da, 0xebfbeb8c, 0x5c85d787, 0x5f330d57, 0x8c72ec21, 0x4e5dd417, 0xef281b47, 0x2340aaf9, 0xd70ca050, 0x3d6af200, 0x8fa10920, 0xab55e97e, 0xbf470922, 0x4b058ac9, 0x53826a64, 0x4c061511, 0xb79744c4, 0x6d7c8063, 0x930f66e7, 0x64731bca, 0xa6c3d52c, 0xcb572e8b, 0xc8508591, 0xafc5e986, 0x3e8103f7, 0xd83ddfbd, 0x8266fa22, 0x27b4d545, 0x42956a81, 0x5d87724e, 0x474d8da6, 0xd26243c1, 0x336743fa, 0x6d220023, 0x7fd9583, 0x58d92968, 0x984f7a05, 0x3834a0db, 0xc2859d15, 0xcd5bd243, 0x49c1a601, 0x7425d214, 0x67fb6417, 0x9ac6aa8d, 0xee7db8c, 0xe14471a, 0xddb24ea4, 0x181c4993, 0x50c0347b, 0x6e8b7402, 0xe07c7328, 0x7a482052, 0x40387ea3, 0xb55934e9, 0x74c3f701, 0xe03ce561, 0x76d4db32, 0x4616671, 0xc7b8ef22, 0xd3bec14b, 0xe7bbd065, 0x2f69ab44, 0x88184e42, 0x61dfe41e, 0x24a4012d, 0xe284c75c, 0xba859b92, 0x5a0829b9, 0xa76736cc, 0x6bec2d76, 0xede26849, 0xefe8fd9, 0x53e6f40e, 0xf816ad7c, 0x6e0acb2a, 0x2cc1bab1, 0xbaec582e, 0xf886c51f, 0x6c90ec7c, 0xa1ff77ff, 0xc85be74b, 0x59a1b1a7, 0x447f1bb9, 0x35ff7110, 0x81f0afc8, 0xb73dc083, 0x2eb41b56, 0xea54ac5b, 0x22c80338, 0x761b3309, 0x5041cc89, 0xd8ac9c25, 0x4a6e7646, 0x5e50e0, 0x4cd467d, 0x7b6c3848, 0x95866bc1, 0x61c53468, 0x7cc132d0}
---

[TestTFGen_SplitN_snapshot - 1]
[]uint32{0xf2a9cd69, 0x858c9f6b, 0x895c599e, 0xd986c01a, 0x7fbc5be3, 0x24fbf184, 0xb3c8c7e1, 0x76ca8066, 0x322c5c54, 0x8a466599, 0x1d1922ac, 0x75f142ec, 0x43576c1b, 0x2ee1b64b, 0xf7b9933d, 0xa37a0848, 0xe4e6685b, 0x2cc676bf, 0x3cf03f2b, 0x721c2651, 0x650b3038, 0x30fccd68, 0xe6f88748, 0x4d3fc8d1, 0xcfa6126a, 0x4d8d80aa, 0x9a26f154, 0x6d96fa0, 0x924f3892, 0x74e5b70f, 0xb5f2e7d9, 0x20497869, 0x10a35d86, 0x43945474, 0xcae48a51, 0x36d40aba, 0xfd7e3463, 0x8d9c829f, 0xaed16203, 0x6561b671, 0x6fe302bf, 0x44688519, 0xd7f0bad2, 0xabb59eff, 0xf0703091, 0x6fb7d4db, 0x73e6b1e3, 0x11b8c074, 0x28d32fc8, 0x2263adcf, 0xf30a92cd, 0x6bfeb051, 0x42fd7e26, 0xfcba3066, 0x34301c0e, 0x184017d9, 0xc96a242e, 0xe7061111, 0xf0506ec6, 0x922b3be2, 0xf8ecf237, 0x2afc4bac, 0xd3d67bc3, 0x97157fc9, 0x218f0d5d, 0x90993b15, 0x87a2a776, 0xe8f9fac5, 0x9db1029f, 0x6b4368ad, 0x459ffebf, 0xe203109f, 0x3d579bc0, 0x70455764, 0xb945cb0e, 0x3aabe754, 0x5f599cf8, 0x88e032f5, 0x16acc2e9, 0xf104cee, 0x75f94dcf, 0x1584282, 0xe3fc2265, 0x2e9dc7a1, 0x91e03a4f, 0x71b61306, 0x4addad13, 0x7a777952, 0x61f4336d, 0x353b6a81, 0xd3309096, 0x4089b870, 0xe4b5b2f5, 0x66864c01, 0xdcddb734, 0x1f07d4ed, 0xe0f177b2, 0x3930314f, 0xdf1f46c1, 0x2c3c4a1d}
---
